### PR TITLE
Adding the notebook for dynamic_embedding demo

### PR DESCRIPTION
**Pull request by thushv89**
# Description

This adds a notebook to demonstrate the usage of the dynamic_embedding module in the `tensorflow_recommender_addons` library.

Fixes # (issue)

## Type of change

- [ ] Bug fix
- [x] New Tutorial
- [ ] Updated or additional documentation
- [ ] Additional Testing
- [ ] New Feature

# Checklist:

- [ ] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/recommenders-addons/blob/master/CONTRIBUTING.md#coding-style)
    - [ ] By running yapf
    - [ ] By running clang-format
- [ ] This PR addresses an already submitted issue for TensorFlow Recommenders-Addons
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

# How Has This Been Tested?

N/A

### CODE REVIEW

File: demo/dynamic_embedding/movielens-1m-keras-notebook/movielens_dynamic_embedding_tutorial.ipynb [hunk 1]
```diff
@@ -0,0 +1,1186 @@
+{
```
Comment (rhdong) [inline demo/dynamic_embedding/movielens-1m-keras-notebook/movielens_dynamic_embedding_tutorial.ipynb:1]: <p><code>de.keras.layers.BasicEmbedding</code>  currently has a better alias <code>de.keras.layers.Embedding</code>  which is close to the original keras layer name.</p><p>So it would be better to change all of the BasicEmbedding to Embedding. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/recommenders-addons/pull/315/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

Comment (rhdong) [inline demo/dynamic_embedding/movielens-1m-keras-notebook/movielens_dynamic_embedding_tutorial.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #18.                        int(self.model.lookup_vocab_sizes[k]*0.8), 
</pre><p>The <code>num_reserved</code> looks like too low, which may hurt the AUC significantly, may we change it to <code>0.95</code>?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/recommenders-addons/pull/315/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

**Comment by rhdong** [comment]
Hi @thushv89, thank you for your excellent job. Here is some comments maybe helpful:
- `BasicEmbedding` could be changed to `Embedding`
- I found the embedding size fluctuates too much, which may hurt the AUC of `DE restrict.`  So Setting the 'restrict` configuration `num_reserved` from `0.8` to `0.95` would be better, and re-run testing is needed. This change may help get a higher AUC. (Please confirm the TFRA version is `0.6.0` when running the test.)
Thank you so much!

File: demo/dynamic_embedding/movielens-1m-keras-notebook/movielens_dynamic_embedding_tutorial.ipynb (hunk 1 above)
Comment (thushv89) [reply demo/dynamic_embedding/movielens-1m-keras-notebook/movielens_dynamic_embedding_tutorial.ipynb:1]: Hey @rhdong,

On this, it takes about an hour (if not more) for me to run this end to end (as I need to generate graphs with comparisons). I think it's fine to have a lower accuracy with restrict. 

The way I see, we have proven that `dynamic_embedding` can reach competitive performance with standard `Embedding` when not using `restrict()`. So I think it's okay to have sub-optimal performance using `restrict()` at `0.8`. The goal of the section using `restrict()` wasn't to reach good performance, but to show the effect of `restrict()` functionality. I'm happy to highlight this in the blog if you think that's better. The readers should be able to interpolate between the lines and understand the higher the `num_reserved` the better. I think that's just an implementation detail we don't have to worry too much? 

What do you think?

Comment (rhdong) [reply demo/dynamic_embedding/movielens-1m-keras-notebook/movielens_dynamic_embedding_tutorial.ipynb:1]: Yeah, highlighting it makes sense, thank you so much!

**Review by rhdong — APPROVED** [review]
LGTM
